model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -721,15 +731,39 @@ message StateKey {
     bytes key = 4;
   }
 
+  // Represents a request for the keys of a multimap associated with a specified
+  // user key and window in a PCollection. See
+  // https://s.apache.org/beam-fn-state-api-and-bundle-processing for further
+  // details.
+  //
+  // Can only be used to perform StateGetRequests and StateClearRequests on the
+  // user state.
+  //
+  // For a PCollection<KV<K, V>>, the response data stream will be a
+  // concatenation of all K's associated with the specified user key and window.
+  // See https://s.apache.org/beam-fn-api-send-and-receive-data for further
+  // details.
   message MultimapKeysUserState {
     // (Required) The id of the PTransform containing user state.
     string transform_id = 1;
     // (Required) The id of the user state.
     string user_state_id = 2;
     // (Required) The window encoded in a nested context.
     bytes window = 3;
+    // (Required) The key of the currently executing element encoded in a
+    // nested context.
+    bytes key = 4;
   }
 
+  // Represents a request for the values of the lookup key associated with a
+  // specified user key and window in a PCollection. See
+  // https://s.apache.org/beam-fn-state-api-and-bundle-processing for further
```

Comment:
Ditto for not being a PCollection but data that is associated with the transform itself.

See https://beam.apache.org/blog/stateful-processing/